test/core/derivatives/implementation/autograd.py

```diff
@@ -42,41 +42,28 @@ def jac_t_vec_prod(self, vec):  # noqa: D102
     def jac_t_mat_prod(self, mat):  # noqa: D102
         return stack([self.jac_t_vec_prod(vec) for vec in mat])
 
-    def weight_jac_t_mat_prod(self, mat, sum_batch, subsampling=None):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "weight", mat, sum_batch, subsampling=subsampling
-        )
-
-    def bias_jac_t_mat_prod(self, mat, sum_batch, subsampling=None):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "bias", mat, sum_batch, subsampling=subsampling
-        )
-
-    def bias_ih_l0_jac_t_mat_prod(self, mat, sum_batch, subsampling=None):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "bias_ih_l0", mat, sum_batch, axis_batch=1, subsampling=subsampling
-        )
-
-    def bias_hh_l0_jac_t_mat_prod(self, mat, sum_batch, subsampling=None):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "bias_ih_l0", mat, sum_batch, axis_batch=1, subsampling=subsampling
-        )
-
-    def weight_ih_l0_jac_t_mat_prod(
-        self, mat, sum_batch, subsampling=None
-    ):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "weight_ih_l0", mat, sum_batch, axis_batch=1, subsampling=subsampling
-        )
-
-    def weight_hh_l0_jac_t_mat_prod(
-        self, mat, sum_batch, subsampling=None
-    ):  # noqa: D102
-        return self.param_jac_t_mat_prod(
-            "weight_hh_l0", mat, sum_batch, axis_batch=1, subsampling=subsampling
+    def param_mjp(
+        self,
+        param_str: str,
+        mat: Tensor,
+        sum_batch: bool,
+        subsampling: List[int] = None,
+    ) -> Tensor:  # noqa: D102
+        axis_batch = 0 if param_str in ("weight", "bias") else 1
```

Comment:
Use [`get_batch_axis`](https://github.com/f-dangel/backpack/blob/development/backpack/utils/subsampling.py#L34-L49) to infer batch dimension.